apps/omg_watcher/lib/omg_watcher/block_validator.ex

```diff
@@ -48,7 +57,7 @@ defmodule OMG.Watcher.BlockValidator do
     end
   end
 
-  @spec verify_transactions(transactions :: list(Transaction.Recovered.t())) ::
+  @spec verify_transactions(transactions :: list(Transaction.Signed.tx_bytes())) ::
           {:ok, list(Transaction.Recovered.t())}
           | {:error, Transaction.Recovered.recover_tx_error()}
   defp verify_transactions(transactions) do
```

Comment:
Didn't check all the logic, but are you sure the txbytes that you receive here are signed (included signatures in their RLP array)?

Reply:
Ahh, seems that you have them signed yeah: https://github.com/omgnetwork/elixir-omg/blob/master/apps/omg/lib/omg/block.ex#L48
So the hash is calculated using unsigned bytes but you have the signed bytes in the block object as well

Reply:
So the question is, do you need the signed bytes or not? Maybe we can save a few Mb per block using unsigned tx bytes

Reply:
hey @mederic-p. Thanks. 

The Merkle root will be constructed from the raw transaction bytes, but the transaction bytes in the list will be signed. 

As we would like to validate that transactions are correctly signed, we need the signed bytes. 

(... for now validation is stateless, so checking that signature is correctly formed and a signature exists for each input)